src/SFA.DAS.Employer.PR.Web/Controllers/Requests/DeclineCreateAccountController.cs

```diff
@@ -1,33 +1,58 @@
 ﻿using Microsoft.AspNetCore.Authorization;
 using Microsoft.AspNetCore.Mvc;
 using SFA.DAS.Employer.PR.Domain.Interfaces;
-using SFA.DAS.Employer.PR.Domain.OuterApi.Responses;
 using SFA.DAS.Employer.PR.Web.Constants;
+using SFA.DAS.Employer.PR.Web.Extensions;
 using SFA.DAS.Employer.PR.Web.Infrastructure;
+using SFA.DAS.Employer.PR.Web.Infrastructure.Services;
 using SFA.DAS.Employer.PR.Web.Models.Requests;
+using SFA.DAS.Employer.PR.Web.Models.Session;
 
 namespace SFA.DAS.Employer.PR.Web.Controllers.Requests;
 
-
-[Route("requests")]
-public class DeclineCreateAccountController(IOuterApiClient _outerApiClient) : Controller
+[Authorize]
+[Route("request/{requestId:guid}/createaccount/decline", Name = RouteNames.DeclineCreateAccount)]
+public class DeclineCreateAccountController(IOuterApiClient _outerApiClient, ISessionService _sessionService) : Controller
 {
-    [Authorize]
     [HttpGet]
-    [Route("{requestId:guid}/createaccount/decline", Name = RouteNames.DeclineCreateAccount)]
-    public async Task<IActionResult> Index(Guid requestId, CancellationToken cancellationToken)
+    public IActionResult Index(Guid requestId, CancellationToken cancellationToken)
     {
         Request.HttpContext.Items.Add(SessionKeys.AccountTasksKey, true);
-        GetPermissionRequestResponse permissionRequest = await _outerApiClient.GetPermissionRequest(requestId, cancellationToken);
+
+        var sessionModel = _sessionService.Get<AccountCreationSessionModel>();
+        if (string.IsNullOrEmpty(sessionModel?.ProviderName))
+        {
+            return RedirectToRoute(RouteNames.CreateAccountCheckDetails, new { requestId });
+        }
 
         var backLink = Url.RouteUrl(RouteNames.CreateAccountCheckDetails, new { requestId });
 
         var vm = new DeclineCreateAccountViewModel
         {
-            ProviderName = permissionRequest.ProviderName.ToUpper(),
+            ProviderName = sessionModel.ProviderName.ToUpper(),
```

Comment:
Are you doubling up ToUpper in View as well?

Reply:
yes. I will remove from view,